src/Extensions/S3/EgressArtifactResult.cs

```diff
@@ -0,0 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace Microsoft.Diagnostics.Monitoring.S3
+{
+    [DebuggerDisplay("{Succeeded?\"Succeeded\":\"Failed\",nq}: {Succeeded?ArtifactPath:FailureMessage}")]
+    internal sealed class EgressArtifactResult
```

Comment:
For all of these files that are copies from the other extension, please:
- Create a new project (e.g. Microsoft.Diagnostics.Monitoring.Extension.Common)
- Add these classes to that project; make sure all of the types are internal
- Reference this project in each of the extension projects.

Reply:
Do you want this as part of this PR, or in a follow-up? In theory it shouldn't be too much extra since it's just renamespacing and moving some files, but I want to make sure this one doesn't become too unwieldy from a review perspective.

Reply:
I think you should do it as part of this PR; there won't be as many new files because they'll look like moves with minor changes in them